tests/pems/crl/generate.sh

```diff
@@ -0,0 +1,203 @@
+sign_cert ./leaf/leaf_revoked intermediate
+popd 1> /dev/null || exit
+
+pushd ./root/intermediate_revoked 1> /dev/null || exit
+sign_cert ./leaf/leaf intermediate_revoked
+sign_cert ./leaf/leaf_revoked intermediate_revoked
+popd 1> /dev/null || exit
+
+# Generate CRLs and revoke the revoked certificates
+pushd ./root 1> /dev/null || exit
+generate_crl root
+revoke_certificate ./intermediate_revoked/intermediate_revoked.crt root
+popd 1> /dev/null || exit
+
+pushd ./root/intermediate 1> /dev/null || exit
+generate_crl intermediate
+revoke_certificate ./leaf/leaf_revoked.crt intermediate
+popd 1> /dev/null || exit
+
+pushd ./root/intermediate_revoked 1> /dev/null || exit
+generate_crl intermediate_revoked
+revoke_certificate ./leaf/leaf_revoked.crt intermediate_revoked
+popd 1> /dev/null || exit
+
+#
+# Generate invalid CRLs
+#
+
+pushd ./root/intermediate 1> /dev/null || exit
+
+# CRL with invalid thisUpdate field
+# - requires openssl 3.0 to generate
+ca_name=intermediate openssl ca -config "${OPENSSL_CONF_FILE}" -gencrl \
+  -crl_lastupdate 21220101000000Z \
+  -keyfile intermediate.key \
+  -cert intermediate.crt \
+  -out intermediate_invalid_thisUpdate.crl.pem \
+  || exit
+openssl crl -inform PEM \
+  -in intermediate_invalid_thisUpdate.crl.pem \
+  -outform DER -out intermediate_invalid_thisUpdate.crl \
+  || exit
+rm intermediate_invalid_thisUpdate.crl.pem
+
+# CRL with invalid nextUpdate field
+# - requires openssl 3.0 to generate
+ca_name=intermediate openssl ca -config "${OPENSSL_CONF_FILE}" -gencrl \
+  -crl_nextupdate 19220101000000Z \
+  -keyfile intermediate.key \
+  -cert intermediate.crt \
+  -out intermediate_invalid_nextUpdate.crl.pem \
+  || exit
+openssl crl -inform PEM \
+  -in intermediate_invalid_nextUpdate.crl.pem \
+  -outform DER -out intermediate_invalid_nextUpdate.crl \
+  || exit
+rm intermediate_invalid_nextUpdate.crl.pem
+
+popd 1> /dev/null || exit
+
+#
+# Create certificate chain pems and copy CRLs
+#
+
+cat ./root/intermediate/leaf/leaf.crt \
+  ./root/intermediate/intermediate.crt \
+  ./root/root.crt \
+  > chain_root_valid_valid.pem
+
+cat ./root/intermediate/leaf/leaf_revoked.crt \
+  ./root/intermediate/intermediate.crt \
+  ./root/root.crt \
+  > chain_root_valid_revoked.pem
+
+cat ./root/intermediate_revoked/leaf/leaf.crt \
+  ./root/intermediate_revoked/intermediate_revoked.crt \
+  ./root/root.crt \
+  > chain_root_revoked_valid.pem
+
+cat ./root/intermediate_revoked/leaf/leaf_revoked.crt \
+  ./root/intermediate_revoked/intermediate_revoked.crt \
+  ./root/root.crt \
+  > chain_root_revoked_revoked.pem
+
+cp ./root/root.crl ./root.crl
+cp ./root/intermediate/intermediate.crl ./intermediate.crl
+cp ./root/intermediate_revoked/intermediate_revoked.crl ./intermediate_revoked.crl
+
+cp ./root/intermediate/intermediate_invalid_thisUpdate.crl ./intermediate_invalid_thisUpdate.crl
+cp ./root/intermediate/intermediate_invalid_nextUpdate.crl ./intermediate_invalid_nextUpdate.crl
+
+# Cleanup
+rm -r ./root
```

Comment:
Can we use a temporary directory instead of root? `mktemp -d`

Reply:
Added: https://github.com/aws/s2n-tls/blob/c1615a6fcf518ac9d84564d9647d97029f3106b2/tests/pems/crl/generate.sh#L173-L174